python_modules/dagster/dagster/_core/definitions/events.py

```diff
@@ -690,3 +699,35 @@ def __new__(cls, hook_name: str, is_skipped: Optional[bool] = None):
 
 
 UserEvent = Union[AssetMaterialization, AssetObservation, ExpectationResult]
+
+
+def validate_asset_event_tags(tags: Optional[Mapping[str, str]]) -> Optional[Mapping[str, str]]:
+    from dagster._core.definitions.utils import validate_tag_strict
+
+    from .data_version import (
+        CODE_VERSION_TAG,
+        DATA_VERSION_TAG,
+        INPUT_DATA_VERSION_TAG_PREFIX,
+        INPUT_EVENT_POINTER_TAG_PREFIX,
+    )
+
+    if tags is None:
+        return None
+
+    for key, value in tags.items():
+        # The format of these particular tags does not fit strict validation:
+        # - The keys have two slashes
+        # - The value for the data version tag can be an arbitrary string
+        if key in [CODE_VERSION_TAG, DATA_VERSION_TAG, _OLD_DATA_VERSION_TAG]:
+            continue
+
+        if (
+            key.startswith(INPUT_DATA_VERSION_TAG_PREFIX)
+            or key.startswith(INPUT_EVENT_POINTER_TAG_PREFIX)
+            or key.startswith(_OLD_INPUT_DATA_VERSION_TAG_PREFIX)
+        ):
+            continue
```

Comment:
can we encapsulate this in a helper function and document it nicely? e.g. (`is_system_defined_asset_event_tag`) or something?

Also what is the stated behavior if a user provides their own version of these tags?

Reply:
> Also what is the stated behavior if a user provides their own version of these tags?

It overrides the system-provided tag.  That's actually the motivation for this PR, to allow overriding the provenance tags.

Reply:
Got it.

I think some function like

```python
def is_system_asset_event_tag(key):
  return key in [CODE_VERSION_TAG, DATA_VERSION_TAG, _OLD_DATA_VERSION_TAG]
    or key.startswith(INPUT_DATA_VERSION_TAG_PREFIX)
    or key.startswith(INPUT_EVENT_POINTER_TAG_PREFIX)
    or key.startswith(_OLD_INPUT_DATA_VERSION_TAG_PREFIX)
```

would make this much more clear

Reply:
👍